src/Symfony/Component/Mailer/Transport/AbstractTransport.php

```diff
@@ -76,7 +76,7 @@ public function send(RawMessage $message, Envelope $envelope = null): ?SentMessa
             $envelope = $event->getEnvelope();
             $message = $event->getMessage();
 
-            if ($message instanceof TemplatedEmail && ($message->getTextTemplate() || $message->getHtmlTemplate())) {
+            if ($message instanceof TemplatedEmail && (!$message->getTextBody() && $message->getTextTemplate() || !$message->getHtmlBody() && $message->getHtmlTemplate())) {
```

Comment:
Would this be enough?

```suggestion
            if ($message instanceof TemplatedEmail && (null !== $message->getTextTemplate() || null !== $message->getHtmlTemplate())) {
```

Reply:
I don't think so. Both methods are overwritten in the NotificationEmail class: https://github.com/symfony/symfony/blob/6.2/src/Symfony/Bridge/Twig/Mime/NotificationEmail.php#L158-L174

Reply:
for me

(null !== $message->getTextTemplate() || null !== $message->getHtmlTemplate())
is true, $message->getTextTemplate() is @email/default/notification/body.txt.twig and $message->getHtmlTemplate() is @email/default/notification/body.html.twig

Reply:
Let me have a look.

Reply:
For my issue [https://github.com/symfony/symfony/issues/48472](https://github.com/symfony/symfony/issues/48472) this fixes it:

`if ($message instanceof TemplatedEmail && (!$message->getTextBody() && $message->getTextTemplate() || !$message->getHtmlBody() && $message->getHtmlTemplate())) {`

Reply:
See #48505 for an alternative fix.